src/State/Hub.php

```diff
@@ -199,6 +200,18 @@ public function addBreadcrumb(Breadcrumb $breadcrumb): bool
         return null !== $breadcrumb;
     }
 
+    /**
+     * {@inheritdoc}
+     */
+    public function await(): void
+    {
+        $transport = $this->getClient()->getTransport();
+
+        if ($transport instanceof AsyncTransportInterface) {
```

Comment:
I did this so you won't have to consider what client the user is using. It always works just is a noop if it's not an async client. This prevents a try/catch block or checking the transport yourself and the Hub seemed like a logical place to have this bit since the Hub is well... the Hub.

Reply:
Wouldn't be enough to leave the exposed call on the client an leave it to the user?

Reply:
Leaving the transport exposed you mean?

We definitely could, but I think we should provide an easy method of waiting for async transports without the user needing to know about how that internally works or guard against transports not using that method. 

But if this convenience method is not liked, happy to remove it. Possibly it could be moved elsewhere where it's better suited (on our own client possibly even)?